packages/electrode-archetype-react-app/config/load-dev.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+//
+// Load options from the dev archetype
+//
+
+const Path = require("path");
+const optionalRequire = require("optional-require")(require);
+
+function checkTopDevArchetype(devArchName) {
+  const topPkg = require(Path.resolve("package.json"));
+  if (topPkg.name === devArchName) {
+    // In case electrode-archetype-react-app is being used for test/dev in the -dev archetype
+    // resolve config/archetype in electrode-archetype-react-app-dev's own dir
+    return optionalRequire(Path.resolve("config/archetype"));
+  } else {
+    return optionalRequire(`${devArchName}/config/archetype`);
+  }
+}
+
+//
+// Try to set dev settings, if the dev archetype is available.
+// It may have been removed for production deployment.
+//
+function loadDev(options) {
+  const devOptions = checkTopDevArchetype(options.devArchetypeName);
+  if (devOptions) {
+    Object.assign(options, devOptions);
```

Comment:
Is it ideal that you're manipulating the passed argument directly? (rather than, for instance, Object.assign({}, options, dev.Options) or {...options, ...devOptions}? Is the options variable essentially a modules export? If so, will subsequent requires/imports of that options file receive the altered values?

Reply:
I am not particularly worried about it.  This function is internal part of the whole load options op.  just broken up into files and functions to make the code more logical.